### ISSUE TITLE
[Tab Extended]: QA: Button is not receiving keyboard focus and it is not keyboard accessible.

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9234]
### Description

Page Element: CTA button

Issue Description:
Button is not receiving keyboard focus and it is not keyboard accessible.

Expected:
All the interactive elements should be accessible with keyboard.

User Impact:
When the interactive element in the page is not operable by keyboard, then keyboard only users will not be able to access the element itself. Impacts motor impairment users and screen reader users.

### Component(s) impacted

Tab extended

### Browser

Firefox

### Carbon for IBM.com version

v1.37.0-rc.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

  Web component canary: https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components Web component staging: https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
